src/pyk/kast.py

```diff
@@ -1780,6 +1801,21 @@ def top_down(f: Callable[[KInner], KInner], kinner: KInner) -> KInner:
     return f(kinner).map_inner(lambda _kinner: top_down(f, _kinner))
 
 
+# TODO: make method of KInner
+def var_occurances(term: KInner) -> Dict[str, List[KVariable]]:
```

Comment:
I think `Set` would be a better fit here.

```suggestion
def var_occurances(term: KInner) -> Dict[str, Set[KVariable]]:
```

Reply:
No, I want it to retain duplicates so that `count_vars` works properly.